### DISCUSSION TITLE
Add properly doc comments for some ambiguous type property’s

### DISCUSSION THREAD

**Discussion by howmarketing** [soofstad/react-oauth2-pkce#107] [General]
# 💡Add to ambiguous property helpful DOC comments💡

## ✅ See below an DOC Comment example about helpful doc comment for an ambiguous property

````ts
export type TAuthConfig = {
    clientId: string;
    authorizationEndpoint: string;
    tokenEndpoint: string;
    /**
     * Redirect URI
     *  
     *  @date 7/02/2023 - 4:13:11 PM
     *  
     *  @description
     *  
     *  > - The "redirectUri" property in the "TAuthConfig" type of the "react-oauth2-code-pkce" package.
     *  > - represents the URI to which the user will be redirected after a successful authentication or authorization process with the OAuth2 provider.
     *  > - The "redirectUri" value can be any valid URI, and its format depends on how you have configured your OAuth2 client application with the OAuth2 provider (e.g., Google, Facebook, etc.).
     *  > - Here are some examples of what the "**redirectUri**" could look like:
     *  > - - **Relative URI** _**within the same application**_:
     *  
     *  ````ts
     *  redirectUri: '/login'
     *  ````
     *  
     *  > - - **Absolute URI** _**within the same domain**_:
     *  
     *  ````ts
     *  redirectUri: 'https://example.com/login'
     *  ````
     *  
     *  > - - **External URI** (different domain)::
     *  
     *  ````ts
     *  redirectUri: 'https://external-site.com/callback'
     *  ````
     *  
     *  > - - **URI with query parameters** _(e.g., used for additional data or state handling)_:
     *  
     *  ````ts
     *  redirectUri: 'https://example.com/login?source=app&state=some_state_data'
     *  ````
     *  
     *  > - Remember that the "redirectUri" needs to match one of the redirect URIs that you have registered with the OAuth2 provider when setting up your client application. If the OAuth2 provider supports wildcard URIs or you've explicitly allowed multiple redirect URIs, you may have more flexibility in using different URIs within the same domain.
     *  > - So, in summary, the "redirectUri" can be any valid URI that you've configured in your OAuth2 client application settings. It doesn't have to be limited to the main application or the same domain, as long as it's allowed by the OAuth2 provider and follows the correct format.
     *    
     *  #### _**@type**_
     *  
     *  > - `{string}`
     *  @type {string}
     * 
     *  
     */
    redirectUri: string;
    scope?: string;
    state?: string;
    logoutEndpoint?: string;
    logoutRedirect?: string;
    preLogin?: () => void;
    postLogin?: () => void;
    onRefreshTokenExpire?: (event: TRefreshTokenExpiredEvent) => void;
    decodeToken?: boolean;
    autoLogin?: boolean;
    clearURL?: boolean;
    extraAuthParams?: {
        [key: string]: string | boolean | number;
    };
    extraAuthParameters?: {
        [key: string]: string | boolean | number;
    };
    extraTokenParameters?: {
        [key: string]: string | boolean | number;
    };
    extraLogoutParameters?: {
        [key: string]: string | boolean | number;
    };
    tokenExpiresIn?: number;
    refreshTokenExpiresIn?: number;
    storage?: 'session' | 'local';
};
````

## 👀 SEE BELOW THE DOC COMMENT [VIEW ]👀:

> - S
> - T
> - Y
> - L
> - E

### 💅_(Markdown ref style view)_💅

---

#### _**@date**_

> - 7/02/2023 - 4:13:11 PM;

#### _**@description**_

> - The "redirectUri" property in the "TAuthConfig" type of the "react-oauth2-code-pkce" package.
> - represents the URI to which the user will be redirected after a successful authentication or authorization process with the OAuth2 provider.
> - The "redirectUri" value can be any valid URI, and its format depends on how you have configured your OAuth2 client application with the OAuth2 provider (e.g., Google, Facebook, etc.).
> - Here are some examples of what the "**redirectUri**" could look like:
> - - **Relative URI** _**within the same application**_:

````ts
redirectUri: '/login'
````

> - - **Absolute URI** _**within the same domain**_:

````ts
redirectUri: 'https://example.com/login'
````

> - - **External URI** (different domain)::

````ts
redirectUri: 'https://external-site.com/callback'
````

> - - **URI with query parameters** _(e.g., used for additional data or state handling)_:

````ts
redirectUri: 'https://example.com/login?source=app&state=some_state_data'
````

> - Remember that the "redirectUri" needs to match one of the redirect URIs that you have registered with the OAuth2 provider when setting up your client application. If the OAuth2 provider supports wildcard URIs or you've explicitly allowed multiple redirect URIs, you may have more flexibility in using different URIs within the same domain.
> - So, in summary, the "redirectUri" can be any valid URI that you've configured in your OAuth2 client application settings. It doesn't have to be limited to the main application or the same domain, as long as it's allowed by the OAuth2 provider and follows the correct format.
  
#### _**@type**_

> - `{string}`

**Comment by soofstad** [comment]
Good suggestion! 
I will have a look at implementing standard typescript documentation into the project as soon as I get some time. 

In the mean time, any pull requests are welcome 🙏
